### PR TITLE
feat: configure sonarqube for code coverage analysis

### PR DESCRIPTION
**Pull request by NyH1409**

### CODE REVIEW

File: .github/workflows/ci-with-sonar.yml
Comment (Dinasoa) [inline .github/workflows/ci-with-sonar.yml:1]: Dont create a new file to include sonar in your CI. Just put a placeholder then add the steps in poja-cli if with-sonar is true

**Review by Dinasoa — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml
Comment (Onitsiky) [inline .github/workflows/ci.yml:1]: This change should not event appear here

File: build.gradle [hunk 1]
```diff
@@ -9,12 +9,18 @@ plugins {
     id 'org.openapi.generator' version '5.3.0'
 
     id 'jacoco'
+
```
Comment (Onitsiky) [inline build.gradle:12]: Change using <?sonarqube-java-plugin> for example

File: build.gradle [hunk 2]
```diff
@@ -9,12 +9,18 @@ plugins {
...
+    <?java-plugins>
 }
 
 jacoco {
     toolVersion = "0.8.11"
 }
 
+sonarqube {
```
Comment (Onitsiky) [inline build.gradle:20]: This should only be added when with_sonar is set to true. 
Better to set it in poja-cli and just leave a placeholder here (<?sonarqube-conf> for example)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -33,6 +37,25 @@ jobs:
       - run: chmod +x gradlew
       - run: ./gradlew test
 
+      - name: Cache SonarCloud packages
```
Comment (Onitsiky) [inline .github/workflows/ci.yml:40]: Ibid here

**Review by Onitsiky — APPROVED** [review]

**Pull request merged**